metals/src/main/scala/scala/meta/internal/remotels/RemoteLanguageServer.scala

```diff
@@ -0,0 +1,114 @@
+package scala.meta.internal.remotels
+
+import org.eclipse.{lsp4j => l}
+import scala.meta.internal.metals.UserConfiguration
+import java.{util => ju}
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.JsonParser._
+import scala.util.Try
+import scala.meta.internal.metals.DefinitionResult
+import scala.meta.internal.semanticdb.Scala.Symbols
+import scala.meta.internal.metals.ReferencesResult
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.mtags.MD5
+import com.google.gson.JsonPrimitive
+import com.google.gson.JsonObject
+import org.eclipse.lsp4j.Location
+import scala.meta.internal.metals.BuildTargets
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+
+class RemoteLanguageServer(
+    workspace: () => AbsolutePath,
+    config: () => UserConfiguration,
+    buffers: Buffers,
+    buildTargets: BuildTargets
+)(implicit ec: ExecutionContext) {
+  def isEnabledForPath(path: AbsolutePath): Boolean =
```

Comment:
`isOutsideBuildTargets` ? `isEnabledForPath` is rather misleading as it looks like there is some kind of configuration.

Together with using `Option[RemoteLanguageServer]` it would be the same result, but clearer.

Reply:
Though probably then it's not possible to just change the setting. Maybe just return None in references if `remoteLanguageServer` is not defined?

Reply:
I prefer to keep distinguishing when the remote server is enabled vs. returning empty results. The key motivation to keep the distinction is that I don't want to use the remote language server to be enabled for files that are exported in the build graph.

Reply:
Sure, I just felt it's a bit counter intuitive. `isEnabledForPath` actually checks if file is not in buildTarget. So anything outside the source sets also fits the bill.